File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    n = len(prices)
    answer = [0] * n
    stack = [] 
    
    for i in range(n):
        while stack and prices[stack[-1]] > prices[i]:
            index = stack.pop()
            answer[index] = i - index
        stack.append(i)
        
    while stack:
        index = stack.pop()
        answer[index] = n - index - 1

    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
            
        Deque<Integer> stack = new ArrayDeque<>();
        for(int i = 0; i < prices.length; i++) {
            while(!stack.isEmpty() && prices[stack.peek()] > prices[i]) {
                int index = stack.pop();
                answer[index] = i - index;
            }
            stack.push(i);
        }
        
        while(!stack.isEmpty()) {
            int index = stack.pop();
            answer[index] = prices.length - 1 - index;
        }
        
        return answer;
    }
}